app/src/main/AndroidManifest.xml

```diff
@@ -3,6 +3,7 @@
     package="com.example.lhi06.residue">
 
     <uses-permission android:name="android.permission.INTERNET" />
+    <uses-permission android:name="android.permission.WRITE_INTERNAL_STORAGE"/>
```

Comment:
Have a look at the list of android permissions in the documentation. Can you find this?